src/main/java/org/jabref/gui/fieldeditors/URLUtil.java

```diff
@@ -138,4 +143,17 @@ public static Optional<String> getSuffix(final String link, ExternalApplications
             return Optional.ofNullable(suffix);
         }
     }
+
+    public static URL create(String url) throws MalformedURLException {
+        return URLUtil.createUri(url).toURL();
+    }
+
+    public static URI createUri(String url) {
+        try {
+            String urlFormat = url.replace("|", "%7C");
```

Comment:
Please add a comment explaining why we are doing this and the relevant references.